From dc6d5a44ac4db0196ff4cd8855e427c1e5802af8 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Tue, 12 Nov 2024 11:13:05 +0100
Subject: [PATCH] dmaengine: stm32-dma3: introduce ddata2dev helper

ddata2dev helper returns the device pointer from struct dma_device stored
in stm32_dma3_ddata structure.
Device pointer from struct dma_device has been initialized with &pdev->dev,
so the ddata2dev helper returns &pdev->dev.

Change-Id: Ia58388d80fa74b82f940a5153699258644401759
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/418658
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/dma/stm32/stm32-dma3.c | 38 ++++++++++++++++++++--------------
 1 file changed, 22 insertions(+), 16 deletions(-)

diff --git a/drivers/dma/stm32/stm32-dma3.c b/drivers/dma/stm32/stm32-dma3.c
index 863304450932..f6c0c45a3356 100644
--- a/drivers/dma/stm32/stm32-dma3.c
+++ b/drivers/dma/stm32/stm32-dma3.c
@@ -346,6 +346,11 @@ static struct device *chan2dev(struct stm32_dma3_chan *chan)
 	return &chan->vchan.chan.dev->device;
 }
 
+static struct device *ddata2dev(struct stm32_dma3_ddata *ddata)
+{
+	return ddata->dma_dev.dev;
+}
+
 #ifdef CONFIG_DEBUG_FS
 #include <linux/debugfs.h>
 static void stm32_dma3_dbg_summary_show(struct seq_file *s, struct dma_device *dma_dev)
@@ -375,7 +380,7 @@ static dma_addr_t stm32_dma3_translate_addr(struct stm32_dma3_ddata *ddata, u32
 					    struct device *client, dma_addr_t dma_addr)
 {
 	const struct bus_dma_region *dma_map = client ? client->dma_range_map : NULL;
-	struct device *dev = client ? : ddata->dma_dev.dev;
+	struct device *dev = client ? : ddata2dev(ddata);
 
 	/* No HW translation, no need to check address */
 	if (!ddata->axi_addr_offset)
@@ -609,7 +614,7 @@ static int stm32_dma3_lli_pool_dma_alloc(struct stm32_dma3_ddata *ddata, struct
 
 from_dma_pool:
 	if (ddata->gen_pool)
-		dev_dbg(ddata->dma_dev.dev, "gen_pool not usable, use dma_pool as fallback\n");
+		dev_dbg(ddata2dev(ddata), "gen_pool not usable, use dma_pool as fallback\n");
 
 	/* Allocate LLIs in dma_pool */
 	for (i = 0; i < count; i++) {
@@ -683,7 +688,7 @@ static struct stm32_dma3_swdesc *stm32_dma3_chan_desc_alloc(struct stm32_dma3_ch
 	swdesc->ccr = FIELD_PREP(CCR_LAP, lap);
 
 	/* Set LL base address */
-	base_addr = stm32_dma3_translate_addr(ddata, lap, ddata->dma_dev.dev,
+	base_addr = stm32_dma3_translate_addr(ddata, lap, ddata2dev(ddata),
 					      swdesc->lli[0].hwdesc_addr);
 	writel_relaxed(base_addr & CLBAR_LBA, ddata->base + STM32_DMA3_CLBAR(chan->id));
 	if (ddata->axi_addr_offset && base_addr != swdesc->lli[0].hwdesc_addr)
@@ -765,12 +770,12 @@ static void stm32_dma3_chan_prep_hwdesc(struct stm32_dma3_chan *chan,
 	lap = FIELD_GET(CCR_LAP, swdesc->ccr);
 	if (is_last) {
 		if (is_cyclic)
-			next_lli = stm32_dma3_translate_addr(ddata, lap, ddata->dma_dev.dev,
+			next_lli = stm32_dma3_translate_addr(ddata, lap, ddata2dev(ddata),
 							     swdesc->lli[0].hwdesc_addr);
 		else
 			next_lli = 0;
 	} else {
-		next_lli = stm32_dma3_translate_addr(ddata, lap, ddata->dma_dev.dev,
+		next_lli = stm32_dma3_translate_addr(ddata, lap, ddata2dev(ddata),
 						     swdesc->lli[next].hwdesc_addr);
 	}
 
@@ -1432,7 +1437,7 @@ static int stm32_dma3_alloc_chan_resources(struct dma_chan *c)
 	 */
 	chan->dma_config.direction = DMA_TRANS_NONE;
 
-	ret = pm_runtime_resume_and_get(ddata->dma_dev.dev);
+	ret = pm_runtime_resume_and_get(ddata2dev(ddata));
 	if (ret < 0)
 		return ret;
 
@@ -1453,7 +1458,7 @@ static int stm32_dma3_alloc_chan_resources(struct dma_chan *c)
 	return 0;
 
 err_put_sync:
-	pm_runtime_put_sync(ddata->dma_dev.dev);
+	pm_runtime_put_sync(ddata2dev(ddata));
 
 	return ret;
 }
@@ -1476,7 +1481,7 @@ static void stm32_dma3_free_chan_resources(struct dma_chan *c)
 	if (chan->semaphore_mode)
 		stm32_dma3_put_chan_sem(chan);
 
-	pm_runtime_put_sync(ddata->dma_dev.dev);
+	pm_runtime_put_sync(ddata2dev(ddata));
 
 	/* Reset configuration */
 	memset(&chan->dt_config, 0, sizeof(chan->dt_config));
@@ -1937,11 +1942,11 @@ static bool stm32_dma3_filter_fn(struct dma_chan *c, void *fn_param)
 		if (!(mask & BIT(chan->id)))
 			return false;
 
-	ret = pm_runtime_resume_and_get(ddata->dma_dev.dev);
+	ret = pm_runtime_resume_and_get(ddata2dev(ddata));
 	if (ret < 0)
 		return false;
 	semcr = readl_relaxed(ddata->base + STM32_DMA3_CSEMCR(chan->id));
-	pm_runtime_put_sync(ddata->dma_dev.dev);
+	pm_runtime_put_sync(ddata2dev(ddata));
 
 	/* Check if chan is free */
 	if (semcr & CSEMCR_SEM_MUTEX)
@@ -1963,7 +1968,7 @@ static struct dma_chan *stm32_dma3_of_xlate(struct of_phandle_args *dma_spec, st
 	struct dma_chan *c;
 
 	if (dma_spec->args_count < 3) {
-		dev_err(ddata->dma_dev.dev, "Invalid args count\n");
+		dev_err(ddata2dev(ddata), "Invalid args count\n");
 		return NULL;
 	}
 
@@ -1972,14 +1977,14 @@ static struct dma_chan *stm32_dma3_of_xlate(struct of_phandle_args *dma_spec, st
 	conf.tr_conf = dma_spec->args[2];
 
 	if (conf.req_line >= ddata->dma_requests) {
-		dev_err(ddata->dma_dev.dev, "Invalid request line\n");
+		dev_err(ddata2dev(ddata), "Invalid request line\n");
 		return NULL;
 	}
 
 	/* Request dma channel among the generic dma controller list */
 	c = dma_request_channel(mask, stm32_dma3_filter_fn, &conf);
 	if (!c) {
-		dev_err(ddata->dma_dev.dev, "No suitable channel found\n");
+		dev_err(ddata2dev(ddata), "No suitable channel found\n");
 		return NULL;
 	}
 
@@ -2021,6 +2026,7 @@ static int stm32_dma3_get_axi_port_config(struct platform_device *pdev,
 
 static u32 stm32_dma3_check_rif(struct stm32_dma3_ddata *ddata)
 {
+	struct device *dev = ddata2dev(ddata);
 	u32 chan_reserved, mask = 0, i, ccidcfgr, invalid_cid = 0;
 
 	/* Reserve Secure channels */
@@ -2032,7 +2038,7 @@ static u32 stm32_dma3_check_rif(struct stm32_dma3_ddata *ddata)
 	 * In case CID filtering is not configured, dma-channel-mask property can be used to
 	 * specify available DMA channels to the kernel.
 	 */
-	of_property_read_u32(ddata->dma_dev.dev->of_node, "dma-channel-mask", &mask);
+	of_property_read_u32(dev->of_node, "dma-channel-mask", &mask);
 
 	/* Reserve !CID-filtered not in dma-channel-mask, static CID != CID1, CID1 not allowed */
 	for (i = 0; i < ddata->dma_channels; i++) {
@@ -2052,7 +2058,7 @@ static u32 stm32_dma3_check_rif(struct stm32_dma3_ddata *ddata)
 				ddata->chans[i].semaphore_mode = true;
 			}
 		}
-		dev_dbg(ddata->dma_dev.dev, "chan%d: %s mode, %s\n", i,
+		dev_dbg(dev, "chan%d: %s mode, %s\n", i,
 			!(ccidcfgr & CCIDCFGR_CFEN) ? "!CID-filtered" :
 			ddata->chans[i].semaphore_mode ? "Semaphore" : "Static CID",
 			(chan_reserved & BIT(i)) ? "denied" :
@@ -2060,7 +2066,7 @@ static u32 stm32_dma3_check_rif(struct stm32_dma3_ddata *ddata)
 	}
 
 	if (invalid_cid)
-		dev_warn(ddata->dma_dev.dev, "chan%*pbl have invalid CID configuration\n",
+		dev_warn(dev, "chan%*pbl have invalid CID configuration\n",
 			 ddata->dma_channels, &invalid_cid);
 
 	return chan_reserved;
-- 
2.50.1

